### ISSUE TITLE
Sugestion: Never suggest `{core, std}::intrinsics` items if the crate doesn't have `feature(core_intrinsics)` set.

### ISSUE THREAD

**Issue by Lokathor** [rust-lang/rust-analyzer#14077]
If you have (for example) `size_o` written, and you hit for auto-complete suggestions, then if you *also* have the flag set to prefer core items over std items, you'll get `core::intrinsics::size_of` first in the list, which is essentially never what you want.

**Comment by Veykril** [comment]
related https://github.com/rust-lang/rust-analyzer/issues/3021 https://github.com/rust-lang/rust-analyzer/issues/3020

**Comment by Lokathor** [comment]
Yeah, and this is much more like https://github.com/rust-lang/rust-analyzer/issues/3020 than like https://github.com/rust-lang/rust-analyzer/issues/3021.

**Comment by Lokathor** [comment]
Update: this seems to affect `std` suggestions as well, not just `core`

**Comment by Caellian** [comment]
I find this incredibly annoying. I don't mind suggestions being shown, it's just that they should always rank lower than `std` ones by default (and then checking for `core_intrinsics` isn't necessary (because someone might want to include it at _after_ realizing they need to due to a compiler error from the use statement.

The fact that they almost always end up ranking before `std::mem` (due to alphabetical sorting) means I import the wrong thing 99% of the time and then have to waste 30s for every `std::mem` include.
